.github/workflows/ci.yml

```diff
@@ -62,58 +54,20 @@ jobs:
     strategy:
       fail-fast: false
     steps:
-      - uses: actions/checkout@main
+      - uses: actions/checkout@v3.3.0
       - name: Install rust
-        uses: actions-rs/toolchain@v1
+        uses: actions-rs/toolchain@v1.0.6
         with:
           toolchain: nightly
           override: true
       - name: Test
-        uses: actions-rs/cargo@v1
+        uses: actions-rs/cargo@v1.0.1
```

Comment:
can we unfreeze cargo version? this can easily get outdated. if there is a strict requirement for `v1.0.1`, then we should add a comment here explaining why we need that version

Reply:
This is the recommendation from GH Action's docs. More details on the commit message:

https://github.com/0xPolygonMiden/miden-vm/pull/689/commits/8ce2d9a0e5bbc3b14e3c2338020cf8b253eef8a0

I don't have strong preferences, I was just following the documented best practice. What is the pros of unpinning it?